docs/pages/07_service.adoc

```diff
@@ -0,0 +1,955 @@
+
+transfer LeadTransfer(Lead lead) {
+    field Integer value = lead.value ;
+}
+
+transfer SalesPersonTransfer(SalesPerson salesperson) {
+    field String fullName = salesperson.firstName + " " + salesPerson.lastName ;
+    field LeadTransfer[] leads = salesperson.leads ;
+}
+----
+
+In the example above the `fullName` field of SalesPersonTransfer has default value. The default value is calculated by concatenating the `firstName` and `lastName` fields of SalesPerson mapped entity with a space in the middle. Note the `salesperson` reference in the expression which refers to the mapping field.
+
+The example above defines an additional field for `SalesPersonTransfer`. The field `leads` is defined within SalesPersonTransfer and contains a list of LeadTransfer transfer objects that belong to a particular salesperson. 
+
+=== Auto-mapping
+
+We often encounter the situation when entities need to be delivered to external clients without structural changes. A typical situation of this kind is master data management. To facilitate the creation of such transfer objects, automatic mapping is provided.
+
+If a mapped transfer object is defined without members, all entity fields are automatically mapped in the transfer object with the same names as the names defined in the entity.
+
+----
+entity SalesPerson {
+    field String firstName ;
+    field String lastName ;
+}
+
+transfer SalesPersonTransfer(SalesPerson salesPerson);
+----
+
+In the example above, there will be two fields in the transfer object, the `firstName` and the `lastName`.
+
+Auto-mapping also maps fields with transfer object type. A prerequisite for automatic mapping with a transfer object type is that the referenced transfer object must also already exist as an automatically mapped transfer object. The following example shows a transfer object with a list of transfer objects in one of its fields.
+
+----
+entity SalesPerson {
+    field String firstName ;
+    field String lastName ;
+	relation Lead[] leads ;
+}
+
+entity Lead {
+    field Integer value ;
+}
+
+transfer LeadTransfer(Lead lead);
+transfer SalesPersonTransfer(SalesPerson salesPerson);
+----
+
+[NOTE]
+====
+Note that there can only be one auto-mapped transfer object for an entity.
+====
+
+
+
+== Service
+
+Services are key components of software applications, allowing external systems or users to interact with the application. The services include creating, updating or deleting data or other real-time communications.
+
+Before going into the details of the service declaration, it is necessary to understand the concept of export containing services.
+
+An export is a component that groups services together that are related in some way. Similar to transfer objects, we also distinguish two categories of exports:
+
+* unmapped and
+
+* mapped export.
+
+=== Unmapped export
+
+Unmapped exports are simple components that are solely responsible for grouping services.
+
+To define an unmapped export, use the `export` keyword.
```

Comment:
What is the explanation using the `export` keyword? In the future I will be able to define services that cannot call from outside? I will be able to use that services (not exported) only inside other service?